### PR TITLE
[JENKINS-17706] Prevent Windows cmd to complain when node workspace is given through a UNC path

### PR DESCRIPTION
**Pull request by greydet**

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
I suggest to close this PR. winsw version is available for evaluation purposes

**Comment by daniel-beck** [comment]
@oleg-nenashev Any progress with winsw so this can definitely be closed?

**Comment by oleg-nenashev** [comment]
@daniel-beck Nope, sorry. I've forgotten about winsw for a while :(

**Comment by oleg-nenashev** [comment]
Assigned to myself

File: core/src/main/java/hudson/tasks/BatchFile.java [hunk 1]
```diff
@@ -40,11 +43,27 @@
...
+    	String[] cmdLine = buildCommandLine(script);
+    	
+    	if (ws.getRemote().startsWith("\\\\"))
+    	{
+    		// If workspace path is a UNC path, use pushd/popd to create a temporary drive and avoid cmd.exe to
+    		// complain about unsupported UNC path and moving to C:\Windows as working directory
+	    	String[] prependCmdLine = new String[] {"pushd",  "&&", "set", "WORKSPACE=%CD%", "&&"};
+	    	String[] appendCmdLine = new String[] { "&&", "popd"};
+	    	cmdLine = ObjectArrays.concat(prependCmdLine, cmdLine, String.class);
+	    	cmdLine = ObjectArrays.concat(cmdLine, appendCmdLine, String.class);
```
Comment (jtnord) [inline core/src/main/java/hudson/tasks/BatchFile.java]: > Does this use drive letters

Yes

> and what happens when A-Z are used up?

ca-boom

```
snip
G:\>pushd \\192.168.1.4\home
F:\>pushd \\192.168.1.4\home
E:\>pushd \\192.168.1.4\home
D:\>pushd \\192.168.1.4\home
B:\>pushd \\192.168.1.4\home
' '
CMD does not support UNC paths as current directories.
```

**Comment by oleg-nenashev** [comment]
Closing this PR since WinSW 2.x releases are now publicly available to download. https://github.com/kohsuke/winsw/releases

**Pull request closed without merging**